### PR TITLE
[12.0][ADD] website_sale_stock_force_block: New module to block product sales on website

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa
TT23780
@Tardo @chienandalu Can you review?
There is 2 commits for alow to re-write attrs attribute for custom_message module.

### CODE REVIEW

File: website_sale_stock_available_display/__manifest__.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
     ],
     'data': [
         'views/assets.xml',
-        'views/product_template_views.xml',
```
Comment (chienandalu) [inline website_sale_stock_available_display/__manifest__.py]: Maybe this change should be done in another PR so ocabot can set properly the version, history, etc.

File: website_sale_stock_force_block/README.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+#. Go to *'Sale > Catalog > Product'*.
```
Comment (chienandalu) [inline website_sale_stock_force_block/README.rst]:
```suggestion
#. Go to *'Website > Products > Products*.
```
Sale Management could be not installed

File: website_sale_stock_force_block/README.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+#. Set *Block sales on website and display a message custom* in inventory
+   availability field.
```
Comment (chienandalu) [inline website_sale_stock_force_block/README.rst:44]:
```suggestion
#. Go to *eCommerce* tab.
#. Set *Block sales on website and display a message custom* in inventory
   availability field.
```

File: website_sale_stock_force_block/tests/test_website_sale_stock_force_block.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class websiteSaleStockForceBlock(HttpCase):
+
+    def setUp(self):
+        super().setUp()
+        # Active "Add to cart" option in main products view
+        self.env.ref('website_sale.products_add_to_cart').active = True
+
+        # For testing with website_sale_vat_required module, I avoid
+        # address step to fill partner vat
+        self.env.ref('base.user_admin').partner_id.write({
```
Comment (chienandalu) [inline website_sale_stock_force_block/tests/test_website_sale_stock_force_block.py:15]: Is this necessary for this tests?

File: website_sale_stock_force_block/static/src/js/website_sale_stock_force_block_tour.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    var base = require("web_editor.base");
+
+    var steps = [
+        {
+            trigger: "a:contains('Computer Motherboard')",
+//            extra_trigger: ".product_price:not(:has(a))",
+        },
+        {
+            trigger: "a[href='/shop']",
+            extra_trigger: "#add_to_cart[class='disabled']",
```
Comment (chienandalu) [inline website_sale_stock_force_block/static/src/js/website_sale_stock_force_block_tour.js]: Maybe `"#add_to_cart.disabled"` ?

**Review by chienandalu — CHANGES_REQUESTED** [review]
Funcionality ok. Some remarks:

File: website_sale_stock_force_block/tests/test_website_sale_stock_force_block.py (hunk 4 above)
Comment (sergio-teruel) [reply website_sale_stock_force_block/tests/test_website_sale_stock_force_block.py:15]: Yes.... Other modules can crash this..

File: website_sale_stock_force_block/static/src/js/website_sale_stock_force_block_tour.js (hunk 5 above)
Comment (sergio-teruel) [reply website_sale_stock_force_block/static/src/js/website_sale_stock_force_block_tour.js]: PErfect... thanks

**Comment by sergio-teruel** [comment]
@chienandalu Changes done!!

File: website_sale_stock_available_display/__manifest__.py (hunk 1 above)
Comment (sergio-teruel) [reply website_sale_stock_available_display/__manifest__.py]: Ok... I will open other Pr...

**Review by chienandalu — APPROVED** [review]
Generate the readme again if possible

**Comment by pedrobaeza** [comment]
Red Travis

**Comment by sergio-teruel** [comment]
Ready!!

**Review by chienandalu — APPROVED** [review]

File: website_sale_stock_force_block/models/product_template.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Tecnativa - Sergio Teruel
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    inventory_availability = fields.Selection(selection_add=[
```
Comment (pedrobaeza) [inline website_sale_stock_force_block/models/product_template.py:10]: An uninstall hook that reassigns this value to other standard would be needed.

File: website_sale_stock_force_block/models/res_config_settings.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    inventory_availability = fields.Selection(selection_add=[
```
Comment (pedrobaeza) [inline website_sale_stock_force_block/models/res_config_settings.py:10]: An uninstall hook that reassigns this value to other standard would be needed.

**Review by rafaelbn — APPROVED** [review]
Functionally :+1:

**Comment by pedrobaeza** [comment]
What about the uninstall hook?

**Comment by sergio-teruel** [comment]
@pedrobaeza  Uninstall hook added

**Comment by pedrobaeza** [comment]
Thanks!

/ocabot merge nobump
